### ISSUE TITLE
Geocoder on change input form

### ISSUE THREAD

**Issue by fab-girard** [perliedman/leaflet-control-geocoder#119]
Hello,
thanks for this plugin ! Today, it geocodes on submit form only. I think it could be very interesting to geocode on change input with a delay (aka autocomplete). You could add an option to perform this advanded behavior (and keep default geocode onsubmit).

Let me know what you think about it.

Fabien

**Comment by perliedman** [comment]
Yeah, this is something that should be added. It's pretty straight forward to set up, and some geocoders already support it throught the `suggest` method.

Note however, that some geocoders, Nominatim in particular, does not allow you to use their service for autocomplete.

**Comment by stevage** [comment]
Has anyone made any progress on this? Some geocoder APIs have distinct APIs for autocomplete (eg [HERE Autocomplete](https://developer.here.com/documentation/geocoder-autocomplete/topics/introduction.html), [Mapzen Autocomplete](https://mapzen.com/documentation/search/autocomplete/) ). In some cases (eg HERE), the autocomplete API doesn't actually return coordinates - just possible completed query strings.

**Comment by perliedman** [comment]
Hi @stevage!

I actually had to check the code to see if this is implemented, but actually, it is. So for geocoders that have the `suggest` method, you will get suggestions as you type. It was added in https://github.com/perliedman/leaflet-control-geocoder/commit/26458138c028069da39ad718ea5f6faf142fc75e, and is supported from 1.5.4 and up as far as I can tell.

Just forgot to close this issue, apparently. Thanks for the heads up!
